library/src/scripts/components/search/SearchOption.tsx

```diff
@@ -0,0 +1,60 @@
+/*
+ * @author Stéphane LaFlèche <stephane.l@vanillaforums.com>
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+import * as React from "react";
+import DateTime from "../DateTime";
+import BreadCrumbString from "../BreadCrumbString";
+import { OptionProps } from "react-select/lib/components/Option";
+import { SelectOption } from "../forms/select/overwrites";
+import classNames from "classnames";
+import { IComboBoxOption } from "@library/components/forms/select/SearchBar";
+
+interface IProps extends OptionProps<any> {
+    data: IComboBoxOption<any>;
```

Comment:
Just removing the interface does not improve this being moved to library. It's actually worse now because we have no idea what the data is. We'll need to either:

1. Make the extra data for this a more generic shape and map our multiple possible search endpoints into it.
2. Or Make the search option component pluggable so that KB can supply a different item than forum.

Reply:
https://github.com/vanilla/knowledge/issues/330